### PR TITLE
Sections component complete

### PR DESCRIPTION
**Pull request by F-Ting**
Completed component for displaying TA sections based on chosen course and TA; added minimum styling and navigation back to previous selected components; minor bug fix on given hard-coded URL that was providing the wrong link; added name for route to allow declarative navigation.

### CODE REVIEW

File: js/component-sections.js [hunk 1]
```diff
@@ -1,2 +1,39 @@
...
+    let url = "../get_info.php?course="+to.params.course+"&ta="+to.params.ta
+    fetch(url)
+    .then((res) => res.json())
+                          .then((data) =>
+                                next((vm) => {vm.section_data = data;vm.loading = false}                               ))
+                          .catch((err) => next((vm) => vm.error = err.toString()));
+
+  },
+
+  template: '<div class="sections">'+
```
Comment (siefkenj) [inline js/component-sections.js]: You can use javascript "`" to quote text with newlines in it.

File: css/app.css [hunk 2]
```diff
@@ -8,7 +16,10 @@ body {
 *,
 *:before,
 *:after {
-    box-sizing: inherit;
+  -webkit-box-sizing: border-box;
```
Comment (siefkenj) [inline css/app.css]: Don't use vendor-specific CSS. We will use a transpiler to add all that automatically for us at release time.

File: js/component-sections.js [hunk 3]
```diff
@@ -1,2 +1,39 @@
 // export as global since we're not using a module system
-window.Sections = { template: '<div>Sections {{$route.params.course}} -> {{$route.params.ta}}</div>' }
+window.Sections = Vue.component('Sections',{
+  data(){
```
Comment (siefkenj) [inline js/component-sections.js]: Please format your code with the prettier.js code formatter and set indentation to 4 spaces

Comment (siefkenj) [inline js/component-sections.js]: Even though it's valid JS, I would prefer `data: function() {` instead of `data() {`.

**Review by siefkenj — CHANGES_REQUESTED** [review]

File: css/app.css [hunk 4]
```diff
@@ -55,3 +62,23 @@ navbar h1 {
     padding-right: 5px;
     max-width: 700px;
 }
+.sections{
```
Comment (siefkenj) [inline css/app.css]: Run this through prettier as well. 4 space indentation.

File: js/app.js [hunk 5]
```diff
@@ -11,8 +11,8 @@
 // We'll talk about nested routes later.
 const routes = [
     { path: "/course", component: Courses },
-    { path: "/course/:course", component: TAs },
-    { path: "/course/:course/ta/:ta", component: Sections }
+    { path: "/course/:course", name: "TAs", component: TAs},
```
Comment (siefkenj) [inline js/app.js]: prettier

File: js/component-tas.js [hunk 6]
```diff
@@ -1,2 +1,3 @@
 // export as global since we're not using a module system
+
```
Comment (siefkenj) [inline js/component-tas.js:2]: Don't change files you're not working on. Make sure to review what files have changed before you commit (`git diff`).

File: js/component-sections.js [hunk 7]
```diff
@@ -1,2 +1,49 @@
...
+                    vm.section_data = data;
+                    vm.loading = false;
+                })
+            )
+            .catch(err => next(vm => (vm.error = err.toString())));
+    },
+
+    template:`
+        <div class="sections">
+        <div class="loading" v-if="loading">
```
Comment (siefkenj) [inline js/component-sections.js]: Indentation for HTML, please.

**Review by siefkenj — CHANGES_REQUESTED** [review]

File: .gitignore [hunk 8]
```diff
@@ -0,0 +1 @@
+db/
```
Comment (siefkenj) [inline .gitignore]: Your .gitignore should not be committed

File: get_info.php [hunk 9]
```diff
@@ -67,4 +67,4 @@ function arr_get($array, $key, $default = null){
 }
 
     print('{ "TYPE": "error" }');
-?>
```
Comment (siefkenj) [inline get_info.php:70]: Don't change this file

**Review by siefkenj — COMMENTED** [review]
You should probably copy your files, reclone, and then copy them back so you don't have a bunch of committed files that you shouldn't have and your history is a bit cleaner.

File: get_info.php (hunk 9 above)
Comment (F-Ting) [reply get_info.php:70]: sorry I added a space by accident

**Pull request closed without merging**